### PR TITLE
feat(log): add node inspection function

### PR DESCRIPTION
**Pull request by Akmadan23**
I often use this little function, exposing it in the api might be useful to someone.

### CODE REVIEW

**Review by alex-courtis — CHANGES_REQUESTED** [review]
I like the idea - it will be most useful to have something that I always end up adding log lines for.

I am a bit hesitant:
1. Inspecting nodes is _massive_ as it goes up and down the tree.
2. I don't want to encourage users to look into nodes. They are private and definitely shouldn't be mutated.

How about:
* Change this to log raw instead of print
* Accept an `options` parameter to pass directly to vim.inspect

**Comment by alex-courtis** [comment]
RE 2. I wish we'd never let the nodes escape into public API. I'd like to provide a sanitised copy of nodes like `sort.sorter` however that's going to be difficult...

**Comment by Akmadan23** [comment]
I understand your concerns

> Change this to log raw instead of print

This can be a good idea, but I don't know if it should go in the API, I'd rather put it in the `log` module...

**Comment by alex-courtis** [comment]
> This can be a good idea, but I don't know if it should go in the API, I'd rather put it in the `log` module...

Ah, that would work nicely. I imagine myself doing something like this in an event or mapping:
```lua
local log = require("nvim-tree.log")
log.node("dev", node, "Event.NodeRenamed %s", data.new_name)
```

I'm happy with your best judgement. This will be useful.

**Review by alex-courtis — APPROVED** [review]
pre-emptive approval

File: lua/nvim-tree/log.lua
Comment (Akmadan23) [inline lua/nvim-tree/log.lua:67]: Since the inspect options are rarely changed, I chose to add this little function to set them instead of another parameter to the `node` function.

File: lua/nvim-tree/log.lua
Comment (alex-courtis) [inline lua/nvim-tree/log.lua:79]: That is very handy...

**Review by alex-courtis — APPROVED** [review]
Works beautifully, many thanks.

Do you want me to just merge after approval or do you want to do it?

**Comment by Akmadan23** [comment]
> Do you want me to just merge after approval or do you want to do it?

Oh, it's really no different to me, just merge whenever you feel like the PR is ready

**Comment by alex-courtis** [comment]
> > Do you want me to just merge after approval or do you want to do it?
> 
> Oh, it's really no different to me, just merge whenever you feel like the PR is ready

I am also happy either way.

**Pull request merged**
